spring-kafka/src/main/java/org/springframework/kafka/annotation/KafkaHandlerParser.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2016-2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.annotation;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.Optional;
+import java.util.Set;
+
+import org.springframework.core.MethodIntrospector;
+import org.springframework.core.annotation.AnnotatedElementUtils;
+import org.springframework.util.ReflectionUtils;
+import org.springframework.util.ReflectionUtils.MethodFilter;
+
+/**
+ * Find the appropriate handler method when the target class has a class-level {@link KafkaListener}
+ * annotation and contains a single public method without a {@link KafkaHandler} annotation.
+ *
+ * @author Sanghyeok An
+ *
+ * @since 4.0
+ *
+ * @see KafkaListenerAnnotationBeanPostProcessor
+ */
+public class KafkaHandlerParser {
```

Comment:
This logic doesn't bring any value to be a new class.
I think ` MethodIntrospector.selectMethods` logic in the `KafkaListenerAnnotationBeanPostProcessor` has to be just adjusted to check for `@KafkaHandler` or `@KafkaListener` on methods or the method is single and doesn't matter what annotations on it.